extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/IntervalUtils.java

```diff
@@ -61,4 +64,23 @@ public static List<Interval> difference(final List<Interval> list1, final List<I
 
     return retVal;
   }
+
+  /**
+   * This method checks if the provided interval is aligned by the granularity or is an instance of {@link Intervals#ETERNITY}
+   * ALL granularity isn't aligned to any interval, however this method is defines that ALL granularity matches
+   * an interval with boundary ({@code DateTimes.MIN}, {@code DateTimes.MAX})
+   * This is used to check if the granularity allocation made by the overlord is the same as the one requested in the
+   * SQL query
+   */
+  public static boolean isEternityOrDoesIntervalAlignWithGranularity(
+      final Interval interval,
+      final Granularity granularity
+  )
+  {
+    // AllGranularity needs special handling since AllGranularity#bucketStart always returns false
```

Comment:
I suppose we could fix `AllGranularity.isAligned()` to return true when the interval is eternity.

Reply:
I was not confident if that would break any pre-existing logic. Also, Javadoc states:
```